packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -315,23 +323,26 @@ class _WindowsUtils extends OperatingSystemUtils {
 
   @override
   void zip(Directory data, File zipFile) {
-    final Archive archive = Archive();
-    for (final FileSystemEntity entity in data.listSync(recursive: true)) {
-      if (entity is! File) {
-        continue;
-      }
-      final File file = entity as File;
-      final String path = file.fileSystem.path.relative(file.path, from: data.path);
-      final List<int> bytes = file.readAsBytesSync();
-      archive.addFile(ArchiveFile(path, bytes.length, bytes));
-    }
-    zipFile.writeAsBytesSync(ZipEncoder().encode(archive), flush: true);
+    _processManager.runSync(<String>[
+      _activePowershell,
+      'Compress-Archive',
+      '-Path',
+      data.path,
+      '-DestinationPath',
+      zipFile.path,
+    ]);
   }
 
   @override
   void unzip(File file, Directory targetDirectory) {
-    final Archive archive = ZipDecoder().decodeBytes(file.readAsBytesSync());
-    _unpackArchive(archive, targetDirectory);
+    _processManager.runSync(<String>[
```

Comment:
The Unix versions throw an exception if the process exits with a non-zero status. Two questions:
1. Should we spawn these the same way with `throwOnError: true`.
2. If the expansion fails, will the powershell process fail with an error?

Reply:
Good question, I'll need to test this locally to verify.

Reply:
The powershell process exits with 0, but we can check that the stderr output is non-empty